tests/smartswitch/platform_tests/test_platform_dpu.py

```diff
@@ -51,34 +54,31 @@ def test_midplane_ip(duthosts, enum_rand_one_per_hwsku_hostname, platform_api_co
     pytest_assert(ping_status == 1, "Ping to one or more DPUs has failed")
 
 
-def test_reboot_cause(duthosts, enum_rand_one_per_hwsku_hostname,
+@pytest.mark.disable_loganalyzer
+def test_reboot_cause(duthosts, dpuhosts,
+                      enum_rand_one_per_hwsku_hostname,
                       platform_api_conn, num_dpu_modules):    # noqa: F811
     """
     @summary: Verify `Reboot Cause` using parallel execution.
     """
     duthost = duthosts[enum_rand_one_per_hwsku_hostname]
 
-    dpu_names = [
-        module.get_name(platform_api_conn, index)
-        for index in range(num_dpu_modules)
-    ]
+    ip_address_list, dpu_on_list, dpu_off_list = pre_test_check(
+                                                 duthost,
+                                                 platform_api_conn,
+                                                 num_dpu_modules)
 
     logging.info("Shutting DOWN the DPUs in parallel")
-    dpus_shutdown_and_check(duthost, dpu_names, num_dpu_modules)
+    dpus_shutdown_and_check(duthost, dpu_on_list, num_dpu_modules)
 
     logging.info("Starting UP the DPUs in parallel")
-    dpus_startup_and_check(duthost, dpu_names, num_dpu_modules)
-
-    with SafeThreadPoolExecutor(max_workers=num_dpu_modules) as executor:
-        logging.info("Verify Reboot cause of all DPUs in parallel")
-        for dpu_name in dpu_names:
-            executor.submit(
-                wait_until, DPU_MAX_TIMEOUT, DPU_TIME_INT, 0,
-                check_dpu_reboot_cause, duthost, dpu_name,
-                "Switch rebooted DPU"
-            )
+    dpus_startup_and_check(duthost, dpu_on_list, num_dpu_modules)
+    post_test_dpus_check(duthost, dpuhosts,
+                         dpu_on_list, ip_address_list,
+                         num_dpu_modules, "Switch rebooted DPU")
```

Comment:
Hi @nissampa for Nvidia smartswitch the reboot cause is "Non-Hardware" here, could you add a dict and get the reboot cause based on the platform or vendor?